controllers/provisioning_controller.go

```diff
@@ -120,6 +120,63 @@ func (r *ProvisioningReconciler) readProvisioningCR(req ctrl.Request) (*metal3io
 	return instance, nil
 }
 
+//Ensure Finalizer is present on the Provisioning CR when not deleted and
+//delete resources and remove Finalizer when it is
+func (r *ProvisioningReconciler) checkForCRDeletion(baremetalConfig *metal3iov1alpha1.Provisioning) error {
```

Comment:
i still think you need to return a bool here to say whether or not you have deleted the resources,
and if you have don't continue with the Reconcile .

Reply:
Done.